### PR TITLE
NTSL Improvement: Reintroducing Sig[ma] Tech (Late night regret edition)

### PR DESCRIPTION
**Pull request by H0rologium**
## About The Pull Request

There are two main elements this PR focuses on. I worked on this all night so I am going to sleep for 13 hours after posting this.

For a tl;dr: This PR reintroduces the old signal technician job with some tweaks and additionally fixes some doors on a couple of the maps.

## Why It's Good For The Game - Maps
There are a couple of other parts of the maps updated - these were autogenerated by the hooks installed through the repo. I recompiled the files afterwards to no issues / errors but this is SS13 code we are talking about so im not gonna break down into tears of joy _just yet._ 

My manual changes focused around the doors and terminals leading into Telecommunications. They had permissions set where only someone with permission level 19 (head access [captain and secretary]) could open/access the entity. In most cases adding 61 (tcomms permission) to the correct requirement fields fixed this. This would let tcomms users actually access their job station. Antags still need to find AA or steal IDs/hack to circumvent this. Secretary has been left unchanged (used the OR operator for the door access codes).

## Why It's Good for the Game - Job

Firstly, I originally thought that the removal of sig_tech made sense. I have taken a long break from SS13 to pursue college and my career, but of course my addiction never leaves. I need an excuse to wear my coder socks, Intel doesn't like how they look.

as I have been playing around on multiple servers, I noticed a lot of instances where:
-someone wanted to play sig_tech (me)
-the role was filled by someone wanting to play the more general role
-round had started so most other desirable roles were taken

Essentially the problem was limited roles for this 'area' of the game. I wanted to play a more specific task (telecomms)  but needed to take Secretary to do so. I wanted to reintroduce signal technician to open up secretary for people who specifically wanted to get the role for the more managerial side. 

## How this could be implemented

Obviously adding another job to the roster needs some consideration. In terms of player agency: I did my best effort to:
-Keep signal technician at a level where it is pretty much just the old sig tech. Run to manage tcomms, help engineering. that's it. Reports to Chief Engineer.
-Ensure that Sig Tech had permissions similar to a generic Station Engineer (plus permission 61 for telecomms). 
-Keep all perms for Secretary (no need to change, secretaries can still access everything they could previously)
-Adding this slot expanding engineering roles by 1. Maybe something that could be adjusted in the future maybe not. Leaving a lot of old signal tech stuff at default value when i feel like it. Code inspired by unused code in ModularBungalow and gameplay I experienced in Yog (~12h as sig tech).

#### I didn't just make another open role, however. 

Currently I (in the second commit) set the time required to unlock this role at 10 hours of civilian role. It uses the standard XP counter so current time stats should be used. I could, however, see this role being a special **whitelisted** role. For a job this niche but with the level of authority that NTSL can offer it would make sense. If this works better for the team, i am willing to add that code in myself assuming I don't need access to the database to make it a whitelisted role (from what I have observed it does not).

## Changelog
:cl:
add: More permission cases for map doors/terminals. 
add: Added sig_tech.dm to the job_types folder
del: Removed old signal technician file in ModularBungalow. It wasn't referenced by anything I could find and it made no sense when all of the other jobs are grouped together.
add: 6 hours of sleep deprivation.
del: Removed 2 years off of my life span.
del: Removed 20% from my 8:00 am classes' grade.
/:cl:
(The update to the admins.txt was me deleting too much whitespace after testing locally)

This PR is part of updates first discussed in #533 where I hope to make this role somewhat enjoyable.

### CODE REVIEW

File: ModularBungalow/altjobtitles/alt_titles.dm [hunk 1]
```diff
@@ -7,10 +7,15 @@
...
 /datum/job/engineer
-	alt_titles = list("Maintenance Technician", "Engine Technician", "Electrician", "Signal Technician")
+	alt_titles = list("Maintenance Technician", "Engine Technician", "Electrician")
 	senior_title = list("Project Co-ordinator")
 	ultra_senior_title = list("Bob the Builder")
 
+/datum/job/signal_tech
+	alt_titles = list("DevOps")
+	senior_title = list("Power Coder")
+	ultra_senior_title = list("CentComm Systems Engineer")
```
Comment (EgorDinamit) [inline ModularBungalow/altjobtitles/alt_titles.dm:17]:
```suggestion
	ultra_senior_title = list("CentCom Systems Engineer")
```

File: ModularBungalow/altjobtitles/alt_titles.dm [hunk 2]
```diff
@@ -7,10 +7,15 @@
 
 //Engineering
 /datum/job/engineer
-	alt_titles = list("Maintenance Technician", "Engine Technician", "Electrician", "Signal Technician")
+	alt_titles = list("Maintenance Technician", "Engine Technician", "Electrician")
 	senior_title = list("Project Co-ordinator")
 	ultra_senior_title = list("Bob the Builder")
 
+/datum/job/signal_tech
+	alt_titles = list("DevOps")
```
Comment (EgorDinamit) [inline ModularBungalow/altjobtitles/alt_titles.dm:15]: Weird job title, not gonna lie.
Make it "Development Operative" instead or something?

**Review by EgorDinamit — CHANGES_REQUESTED** [review]
Change PR body to have one changelog, otherwise it won't parse.

File: ModularBungalow/altjobtitles/alt_titles.dm (hunk 1 above)
Comment (bruhlookatthisdood) [reply ModularBungalow/altjobtitles/alt_titles.dm:17]: wow not making it a meme name like the true ultra senior titles.. shaaaaaame

**Comment by H0rologium** [comment]
> Change PR body to have one changelog, otherwise it won't parse.

Good to know, combined both changelogs.

File: code/modules/jobs/job_types/signal_tech.dm [hunk 3]
```diff
@@ -6,22 +6,23 @@
...
 
-//	display_order = JOB_DISPLAY_ORDER_SIGNAL_TECHNICIAN
+	access = list(ACCESS_ENGINE, ACCESS_ENGINE_EQUIP, ACCESS_GATEWAY, ACCESS_TCOMSAT, ACCESS_MAINT_TUNNELS, ACCESS_MECH_ENGINE, ACCESS_AUX_BASE,
+									ACCESS_EXTERNAL_AIRLOCKS, ACCESS_CONSTRUCTION, ACCESS_ATMOSPHERICS, ACCESS_MINERAL_STOREROOM)
+	minimal_access = list(ACCESS_ENGINE, ACCESS_ENGINE_EQUIP, ACCESS_GATEWAY, ACCESS_TCOMSAT, ACCESS_MAINT_TUNNELS,ACCESS_EXTERNAL_AIRLOCKS, ACCESS_CONSTRUCTION, ACCESS_MINERAL_STOREROOM)
+	paycheck = PAYCHECK_MEDIUM
+	paycheck_department = ACCOUNT_ENG
+	display_order = JOB_DISPLAY_ORDER_STATION_ENGINEER
 	bounty_types = CIV_JOB_ENG
-//	departments = DEPARTMENT_ENGINEERING
```
[the remark this answers is not in the document]
Comment (H0rologium) [reply code/modules/jobs/job_types/signal_tech.dm:22]: Good line to point out! However currently the other engineering jobs do not require this line to be added, leading me to believe it is older code in place.

I had initially tried it out and removing this line seemed to help make things run smoother. I looked at station_engineer.dm and chief_engineer.dm to decide to remove this line as those jobs do not include a "departments" variable.

**Comment by Kitsunemitsu** [comment]
Hey man, I'd like to talk about this, actually.
I like Signal Tech, but it actually came up as an issue with the previous staff team. 
I am currently talking with our new staff about approving and implementing this feature.

**Comment by H0rologium** [comment]
After talking to some of the code managers on the discord, we reached a compromise. The job code has been undone and this PR will shift to focus on the permission fixes on the maps. Feel free to contact me for any other questions.

**Comment by bruhlookatthisdood** [comment]
remove seceratary
revive sigma tech

**Comment by EgorDinamit** [comment]
> remove seceratary revive sigma tech

Both jobs are unnecessary in my opinion.
As for the signal tech you could read my message on the Discord: It's pointless separation. Same thing as if you had a dedidacted role for the nanites/xenobio/toxins instead of simply "scientist".

File: ModularBungalow/altjobtitles/alt_titles.dm [hunk 4]
```diff
@@ -7,10 +7,15 @@
 
 //Engineering
 /datum/job/engineer
-	alt_titles = list("Maintenance Technician", "Engine Technician", "Electrician", "Signal Technician")
+	alt_titles = list("Maintenance Technician", "Engine Technician", "Electrician")
 	senior_title = list("Project Co-ordinator")
 	ultra_senior_title = list("Bob the Builder")
 
+/datum/job/signal_tech
```
Comment (EgorDinamit) [inline ModularBungalow/altjobtitles/alt_titles.dm:14]: Since you removed the job, remove it from there as well.

Comment (EgorDinamit) [reply ModularBungalow/altjobtitles/alt_titles.dm:14]: And re-add the alt title to the engineer.

**Comment by H0rologium** [comment]
I have too much going on right now to reclone the repo again (I have to constantly delete my git folder because I'm out of hard drive space.)
I am just going to close this PR instead of making another modification. Nothing fixed is game breaking and gameplay has remained mostly unaffected by changes proposed here. Someone else is welcome to use the commits for their own PR, but I do not have the capability of continuing this PR for the time being.

**Pull request closed without merging**